docs/usage/operations.rst

```diff
@@ -723,3 +723,133 @@ Class Definitions
+
+    **Why use this SaferAddFieldOneToOne operation?**
+    -------------------------------------------------
+
+    When using Django's default ``AddField`` operation, the SQL created has the
+    following form:
+
+    .. code-block:: sql
+
+      BEGIN;
+      --
+      -- Add field foo to bar
+      --
+      ALTER TABLE "myapp_bar"
+      ADD COLUMN "foo_id" bigint NULL
+      UNIQUE CONSTRAINT "auto_gen_constraint_name"
+      REFERENCES "myapp_foo"("id")
+      DEFERRABLE INITIALLY DEFERRED;
+
+      SET CONSTRAINTS "auto_gen_constraint_name" IMMEDIATE;
+      COMMIT;
+
+
+    The ``ALTER TABLE`` command takes an AccessExclusive lock, which is the
+    highest level of locking. It will block reads and writes on the table.
+    At the same time, Postgres will serially create the constraint while that
+    lock is held, which can potentially take a long time.
+
+    The below are the queries executed by this operation in order to avoid the
+    two problems above:
+
+    .. code-block:: sql
+
+      -- This operation takes an AccessExclusiveLock, but for a very short
+      -- duration. Adding a nullable field in Postgres doesn't require a full
+      -- table scan starting on version 11.
+      ALTER TABLE "myapp_bar" ADD COLUMN IF NOT EXISTS "foo_id" bigint NULL;
+
+      -- This operation takes an ShareUpdateExclusiveLock. It won't block
+      -- reads or writes on the table.
+      SET lock_timeout TO '0';
+      CREATE UNIQUE INDEX CONCURRENTLY IF NOT EXISTS "bar_foo_id_uniq" ON "myapp_bar" ("foo_id");
+      SET lock_timeout TO '10s';
+
+      -- This operation takes an AccessExclusiveLock, but for a very short
+      -- duration as it leverages the unique constraint index above to create
+      -- the constraint. By default this constraint will be IMMEDIATE, which
+      -- is what we want.
+      ALTER TABLE "myapp_bar" ADD CONSTRAINT "bar_foo_id_uniq" UNIQUE USING INDEX "bar_foo_id_uniq";
+
+      -- This operation will take a ShareRowExclusive lock on **both** the foo
+      -- table and the bar table. This will not block reads, but it
+      -- will block insert, updates, and deletes. This will only happen for a
+      -- short time, as this operation won't need to scan the whole table.
+      ALTER TABLE "myapp_bar"
+      ADD CONSTRAINT "myapp_bar_foo_id_fk" FOREIGN KEY ("foo_id")
+      REFERENCES "myapp_foo" ("id")
+      DEFERRABLE INITIALLY DEFERRED
+      NOT VALID;
+
+      -- This query will take a ShareUpdateExclusive lock on the foo table
+      -- (does not block reads nor writes), and a RowShare lock on the bar
+      -- table (does not block reads nor writes).
+      ALTER TABLE foo VALIDATE CONSTRAINT fk_post_bar;
+
+    **NOTE**: Additional queries that are triggered by this operation to
+    guarantee idempotency have been omitted from the snippet above. The key
+    take away is that if this migration fails, it can be attempted again and it
+    will pick up from where it has left (reentrancy).
```

Comment:
typo: "from where it has left **off**"

Reply:
6fb244d04d783e449f11ab2297373b64d8c46616